visualisation/grouped_spectra.py

```diff
@@ -0,0 +1,92 @@
+import pandas as pd
+import plotly.graph_objects as go
+import streamlit as st
+
+from processing import save_read
+from processing import utils
+from . import draw
+
+GS = "Grouped spectra"
+RS = "Raman Shift"
+ORG = "Original spectrum"
+RAW = "Raw Data"
+OPT = "Optimised Data"
+NORM = "Normalized"
+OPT_S = "Optimised Spectrum"
+SHIFT = 'Shift spectra from each other'
+
+
+def show_grouped_plot(df, plots_color, template, spectra_conversion_type):
+    global shift, col1
```

Comment:
Nie. Nigdy.
zmienne globalne to zło w czystej postaci :imp: 
 Nie używamy.
Możesz je po prostu przekazać do tej funkcji.
A w przypadku `col1` lepiej wywołać tę funkcję w kontekście `col1`
czyli **przy wywołaniu** robisz:
```
with col1:
    show_grouped_plot(cośtam, cośtam)
```

Reply:
to nie bedzie takie latwe, trzebaby caly kod przerobic, zeby show_grouped_plot() robil male rzeczy ;]
bo tak to mam wewnatrz wyswietlanie widm slajderow itp to jak teraz to dam na kolumny 2 to bedzie z dupy bo w kolumnach beda kolumny i nie wiem, cyz w ogole streamlit jest gotowy na zagniezdzanie kolumn